ChronoVisor/src/ClientRegistryManager.cpp

```diff
@@ -22,24 +22,100 @@ ClientRegistryManager::~ClientRegistryManager() {
     delete clientRegistry_;
 }
 
+ClientInfo& ClientRegistryManager::get_client_info(const std::string &client_id) {
+    std::lock_guard<std::mutex> clientRegistryLock(g_clientRegistryMutex_);
+    auto clientRegistryRecord = clientRegistry_->find(client_id);
+    if (clientRegistryRecord != clientRegistry_->end()) {
+        return clientRegistryRecord->second;
+    }
+}
+
+int ClientRegistryManager::add_story_acquisition(const std::string &client_id, uint64_t &sid, Story *pStory) {
+    std::lock_guard<std::mutex> clientRegistryLock(g_clientRegistryMutex_);
```

Comment:
i think it would make more sense to first check if the story is already acquired by the client and not go to trouble of creating another Story object.

Reply:
> i think it would make more sense to first check if the story is already acquired by the client and not go to trouble of creating another Story object.

The check is firstly done in ChronicleMetaDirectory::acquire_story(), line 282-290. And line 37-40 also checks if sid is in the acquiredStoryList_ of ClineInfo for this client. So I don't get your point here.